### PR TITLE
A note on soft / hard delete

### PR DESCRIPTION
**Pull request by ignacy**
Fixes https://github.com/mdyd-dev/partner-portal/issues/883

A note on soft/hard delete

### CODE REVIEW

File: app/views/pages/tutorials/models/crud-models-using-ajax.liquid [hunk 1]
```diff
@@ -277,6 +277,15 @@ updateForm.addEventListener("submit", Update);
 
 The delete operation also needs `method` and `id` because underneath it all, it is just an update.
 
+#### A note on soft and hard delete
+
+By default when you delete a record what actually happens is this: we set the value for `deleted_at` timestamp and we don't show this record in results unless you search for it explicitly.
+That is why we have `deleted_at: { exists: false }` filter available in some GraphQL queries. This is called soft delete.
```
Comment (pavelloz) [inline app/views/pages/tutorials/models/crud-models-using-ajax.liquid]: If default is to not show records with deleted_at, then we have `exists: true` for that reason to show them.

But i think it is not default, or at least not everywhere. There have been complaints about it.

@Slashek

File: app/views/pages/tutorials/models/crud-models-using-ajax.liquid [hunk 2]
```diff
@@ -277,6 +277,15 @@ updateForm.addEventListener("submit", Update);
...
+We do that because some workflows need this for safety reasons. For example it's possible to restore records removed by accident.
+We have a 30 day retention period for soft deleted records. That means that after 30 days from `delete_at` timestamp those records will be removed and they will no longer be available for restoration.
+That also means we will not charge you after 30 days for storing those records. If your workflow requires longer retention you should be prepared to offload those records by for example exporting instance
```
Comment (pavelloz) [inline app/views/pages/tutorials/models/crud-models-using-ajax.liquid]:
```suggestion
That also means we will not charge you after 30 days for storing those records. If your workflow requires longer retention you should be prepared to backup those records by for example exporting instance
```

File: app/views/pages/tutorials/models/crud-models-using-ajax.liquid [hunk 3]
```diff
@@ -277,6 +277,15 @@ updateForm.addEventListener("submit", Update);
...
+data and keeping it somewhere safe.
```
Comment (pavelloz) [inline app/views/pages/tutorials/models/crud-models-using-ajax.liquid]:
```suggestion
data (using [`pos-cli export`](/tutorials/data-import-export/export) command) and keeping it somewhere safe.
```

File: app/views/pages/tutorials/models/crud-models-using-ajax.liquid (hunk 1 above)
Comment (Slashek) [reply app/views/pages/tutorials/models/crud-models-using-ajax.liquid]: @pavelloz you are correct that's current behaviour

Comment (ignacy) [reply app/views/pages/tutorials/models/crud-models-using-ajax.liquid]: So I'll update this a bit to state we show those records in some queries but not in others and that you can use `exists: true` filter when appropriate to explicitly choose which one is it but this in fact should be documented for all queries... something like "this query includes soft deleted records by default"

**Comment by diana-lakatos** [comment]
Reviewed and edited. Thank you!

**Comment by diana-lakatos** [comment]
@ignacy @pavelloz I was thinking about this: Shouldn't this information be included in a topic aimed at beginners or a less specific topic? E.g. somewhere in Get Started, or Data Export?

**Comment by ignacy** [comment]
This page is about deleting records so I think this section fits here. That said, we should have similar info in some other places, for me the obvious choice would be to have it in query documentation - it at least should say if the default is to include or not the soft deleted records. I also took a look at the queries we have in our guides and it looks like sometimes we omit the `deleted_at` filter like here https://documentation.platformos.com/tutorials/models/loading-related-models#step-2-load-related-model-for-each-record-not-recommended although if someone would use this code he might be suprised with the result. I guess what realy would be needed is an audit of every query and every place this affects :)

**Comment by diana-lakatos** [comment]
@ignacy Thank you! I will open a separate ticket for the tasks you described.

**Pull request merged**